### ISSUE TITLE
`code-block-tools`: add `on-no-tools-run` for zero tool executions

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#891]
## Summary

Add an `on-no-tools-run` option to Code Block Tools to control what happens when Code Block Tools is enabled but zero tools actually execute during the invocation.

rumdl already warns when `--only-code-block-tools` resolves to a configuration with no tools and when configured tool binaries are missing. This option would cover the remaining runtime case where usable tools are configured and available, but none are executed for any processed fenced code block.

## Proposed behavior

```toml
[code-block-tools]
on-no-tools-run = "warn"
```

| Value | Behavior |
| --- | --- |
| `"warn"` | Complete the run, emit one config warning that no Code Block Tools executed, and exit normally (default) |
| `"ignore"` | Silently allow the run to complete with zero tool executions |
| `"fail"` | Complete processing, report that no Code Block Tools executed, and exit non-zero |
| `"fail-fast"` | Fail as soon as rumdl can determine that no tool can execute; otherwise fail at the end if the execution count remains zero |

The check should be based on actual tool executions, not merely whether tools are configured. It should apply to `rumdl check`, `rumdl check --fix`, and `rumdl fmt`, including `--only-code-block-tools`.

## Usage

```toml
[code-block-tools]
enabled = true
on-no-tools-run = "fail"

[code-block-tools.languages]
python = { lint = ["ruff:check"], format = ["ruff:format"] }
```

Given a document with no applicable fenced code blocks:

```markdown
# Example

Documentation without a Python code block.
```

Run:

```console
$ rumdl check docs/
```

With `on-no-tools-run = "fail"`, the invocation should exit non-zero because no configured Code Block Tool actually executed.

## References

- https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md
- #374
- #829
- #889
